vm/systemSmartContracts/delegationManager.go

```diff
@@ -380,6 +382,13 @@ func (d *delegationManager) saveDelegationContractList(list *DelegationContractL
 	return nil
 }
 
+// SetNewGasCost is called whenever a gas cost was changed
+func (d *delegationManager) SetNewGasCost(gasCost vm.GasCost) {
+	d.mutExecution.Lock()
```

Comment:
We are only locking the mutex for setting the gas lock but we are not setting the mutex when actually using the gas lock in `Execute` method.

Reply:
done